File: Backend/Events/Converters/Journals/Quests/RequisiteConverter.cs
using Backend.Domain.Models.Journals.Quests;
using Backend.Events.DTO.Journals.Quests;

namespace Backend.Events.Converters.Journals.Quests;

public static class RequisiteConverter
{
    public static RequisiteDTO ToDTO(Requisite requisite) => new()
    {
        Id = requisite.Id,
        Value = requisite.Value
    };
}

File: Frontend/src/events/converters/journals/quests/requisite.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { RequisiteDTO } from "@/events/dto/journals/quests/requisite.dto";
import type { Requisite } from "@/models";

export class RequisiteConverter {
  public static convert(requisiteDto: DeepRequired<RequisiteDTO>): Requisite {
    return {
      id: requisiteDto.id,
      isDone: requisiteDto.isDone,
    };
  }
}
